portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +34,98 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+private List<String> comments;
+
+  @Override
+  public void init() {
+    comments = new ArrayList<>();
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();    
+    PreparedQuery results = datastore.prepare(query);
+    List<Comment> commentsList = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long id = entity.getKey().getId();
+        String comment = (String) entity.getProperty("comment");
+        long timestamp = (long) entity.getProperty("timestamp");
+
+        Comment com = new Comment(id, comment, timestamp);
+        commentsList.add(com);
+    }
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(commentsList));
+    
+    // String commentJson = convertToJsonUsingGson(comments);
```

Comment:
Best practice would be to delete the old code you're not using anymore. Otherwise, it's easy for lines like these to stick around for years :)